proto/tabletmanagerdata.proto

```diff
@@ -728,9 +738,88 @@ message CheckThrottlerResponse {
   double threshold = 3;
   // Error indicates an error retrieving the value
   string error = 4;
-  // Message 
+  // Message
   string message = 5;
   // RecentlyChecked indicates that the tablet has been hit with a user-facing check, which can then imply
   // that heartbeats lease should be renwed.
   bool recently_checked = 6;
+
+  message Metric {
+    // Name of the metric
+    string name = 1;
+    // StatusCode is HTTP compliant response code (e.g. 200 for OK)
+    int32 status_code = 2;
+    // Value is the metric value collected by the tablet
+    double value = 3;
+    // Threshold is the throttling threshold the table was comparing the value with
+    double threshold = 4;
+    // Error indicates an error retrieving the value
+    string error = 5;
+    // Message
+    string message = 6;
+    // Scope used in this check
+    string scope = 7;
+  }
+  map<string, Metric> metrics = 7;
+}
+
+message GetThrottlerStatusRequest {
+}
+
+message GetThrottlerStatusResponse {
+  // TabletAlias of probed tablet
+  string tablet_alias = 1;
+  string keyspace = 2;
+  string shard = 3;
+
+  // IsLeader indicates if the tablet is the leader of the shard, ie. the primary
+  bool is_leader = 4;
+  // IsOpen per stateManager
+  bool is_open = 5;
+  // IsEnabled per throttler configuration
+  bool is_enabled = 6;
+  // IsDormant: whether the throttler is dormant, ie has not received any checks in a while
+  // and goes into low-frequency probing mode.
+  bool is_dormant = 7;
+
+  // LagMetricQuery is the query used to check the lag metric, a constant used by the throttler.
+  string lag_metric_query = 8;
+  // CustomMetricQuery is the query used to check the custom metric, supplied by the user.
+  string custom_metric_query = 9;
+  // DefaultThreshold is the threshold used by the throttler for the default metric (lag or custom in single-metric throttlers)
+  double default_threshold = 10;
+  // MetricNameUsedAsDefault is the name of the metric used as the default metric: "lag" or "custom", for backwards compatibility
+  // with single-metric throttlers
+  string metric_name_used_as_default = 11;
+
+  message MetricResult {
+    double value = 1;
+    string error = 2;
+  }
+
+  // AggregatedMetrics is a map of metric names to their values/errors
+  // Names are, for example, "self", "self/lag", "shard/lag", "shard/loadavg", etc.
+  map<string, MetricResult> aggregated_metrics = 12;
+  // MetricThresholds is a map of metric names to their thresholds.
+  map<string, double> metric_thresholds = 13;
+
+  message MetricHealth {
+    vttime.Time last_healthy_at = 1;
+    int64 seconds_since_last_healthy = 2;
```

Comment:
This feels a little redundant to me, but fine as is.

Reply:
Yeah, it's redundant -- but great for metrics! Which is why I added it. This represents really well as a gauge.